### PR TITLE
ginkgo: install on demand via `go get -u`

### PR DESCRIPTION
**Pull request by vrothberg**
Install ginkgo on demand via `go get -u` rather than keeping a copy of its
entire source code in the vendor dirctory.  The main motivation for that
is to make `golangci-lint` happy which is continuously throwing up on
the import of a program (i.e., ginkgo).  The linter is broken and stupid
as it ignores flags to ignore dirs and ignores build tags (at least some
linters do) which is blocking us from updating to newer versions.

Signed-off-by: Valentin Rothberg <rothberg@redhat.com>

@containers/podman-maintainers PTAL

One more stab for updating the linter.

### CODE REVIEW

**Comment by vrothberg** [comment]
/hold
Let's not merge this until we've branched for 3.0

**Comment by vrothberg** [comment]
@containers/podman-maintainers PTAL, 3.0 has been branched

File: go.mod
Comment (saschagrunert) [inline go.mod:41]: I think you can remove that line and run a `go mod tidy`.

Comment (vrothberg) [reply go.mod:41]: Nice catch, thanks!

**Review by saschagrunert — APPROVED** [review]
LGTM

**Comment by mheon** [comment]
Why were we vendoring this, anyways?

LGTM, but CI is angry.

**Comment by vrothberg** [comment]
> Why were we vendoring this, anyways?

We wanted to ship CI dependencies in `./vendor` but golangci-lint really doesn't like importing a program (i.e., main) and is completely ignoring `--skip-dir` and build tags (varying across versions and individual linters).  I couldn't see a way forward without getting that out of golangci-lint's way :(

**Comment by vrothberg** [comment]
Had to re-add `github.com/nxadm/tail v1.4.6 // indirect` to make CI happy

**Comment by vrothberg** [comment]
@Luap99 PTAL :)

**Pull request merged**
